### DISCUSSION TITLE
scope["scheme"] is set to comma-separated list when such a value is present in the X-Forwarded-Proto header

### DISCUSSION THREAD

**Discussion by vanschelven** [Kludex/uvicorn#2102] [Potential Issue] [answered]
The `X-Forwarded-Proto` header is underspecified, but may contain comma-separated values in the wild, see e.g. the [CloudFoundry gorouter](https://docs.cloudfoundry.org/concepts/http-routing.html#X-Forwarded-Proto)

If this is the case, the `scope["scheme"]` will contain such comma-separated values. I _think_ this is probably not what we want, as downstream users expect a single value here.

This is where the copying from the header takes place:

https://github.com/encode/uvicorn/blob/e2a39792cc65e45d9ccc5a3908fc79aaa75b7b12/uvicorn/middleware/proxy_headers.py#L70

Compare how Django does this:

https://github.com/django/django/blob/main/django/http/request.py#L264

This seems to be a rather recent addition, the Django ticket contains more info about other projects with similar fixes.
https://code.djangoproject.com/ticket/33569

**Comment by vanschelven** [comment]
example of downstream problem which is caused by this: https://github.com/getsentry/sentry-python/issues/2370

**Comment by Kludex** [comment, marked as answer]
- No action needed, as per my comment on: https://github.com/encode/uvicorn/pull/2104#issuecomment-1974803330
